Mathlib/Analysis/PSeries.lean

```diff
@@ -17,132 +17,218 @@ converges if and only if so does `∑ k, 2 ^ k f (2 ^ k)`. We prove this test in
+      (add_le_add_right (h_u_mono (le_refl 0) n.zero_le) _ : u 0 + 1 ≤ u n + 1),
+      add_le_add_right (h_u_mono n.zero_le n.le_succ) _]
+  have : ∀ k ∈ Ico (u n + 1) (u (n + 1) + 1), f (u (n + 1)) ≤ f k := fun k hk =>
+    hf (Nat.lt_of_le_of_lt (Nat.succ_le_of_lt (h_pos n)) <| (Nat.lt_succ_of_le le_rfl).trans_le
+      (mem_Ico.mp hk).1) (Nat.le_of_lt_succ <| (mem_Ico.mp hk).2)
   convert sum_le_sum this
   simp [pow_succ, two_mul]
-#align finset.sum_condensed_le' Finset.sum_condensed_le'
-
-theorem sum_condensed_le (hf : ∀ ⦃m n⦄, 1 < m → m ≤ n → f n ≤ f m) (n : ℕ) :
-    (∑ k in range (n + 1), 2 ^ k • f (2 ^ k)) ≤ f 1 + 2 • ∑ k in Ico 2 (2 ^ n + 1), f k := by
-  convert add_le_add_left (nsmul_le_nsmul_of_le_right (sum_condensed_le' hf n) 2) (f 1)
-  simp [sum_range_succ', add_comm, pow_succ, mul_nsmul', sum_nsmul]
-#align finset.sum_condensed_le Finset.sum_condensed_le
 
+theorem sum_schlomilch_le (hf : ∀ ⦃m n⦄, 1 < m → m ≤ n → f n ≤ f m) (h_pos : ∀ n, 0 < u n)
+    (hu : ∀ ⦃m n⦄, 0 ≤ m → m ≤ n → u m ≤ u n) (h_succ_diff: succ_diff_bounded u) :
+    ∃ C > (0 : ℝ≥0), ∀ (n : ℕ), ∑ k in range (n + 1), (u (k + 1) - u k) * f (u k) ≤
+    (u 1 - u 0) * f (u 0) + C * ∑ k in Ico (u 0 + 1) (u n + 1), f k := by
+  have h_nonneg : ∀ n, 0 ≤ f n := fun n => zero_le'
+  obtain ⟨C, h, hCpos, hC⟩ := h_succ_diff
+  lift C to ℝ≥0 using h
+  use C
+  constructor
+  · replace hCpos : C > 0 := ENNReal.coe_lt_coe.1 hCpos
+    apply hCpos
+  intro n
+  rw [sum_range_succ', add_comm]
+  convert add_le_add_left _ ((u 1 - u 0) * f (u 0))
+  have : ∀ k ∈ range n, (u (k + 2) - u (k + 1)) * f (u (k + 1)) ≤
+    C * ((u (k + 1) - u k) * f (u (k + 1))) := by
+      intro k _
+      calc
+      (u (k + 2) - u (k + 1)) * f (u (k + 1)) ≤
+      C * (u (k + 1) - u k) * f (u (k + 1)) :=
+        mul_le_mul_of_nonneg_right (hC k) (h_nonneg (u (k + 1)))
+      _ = C * ((u (k + 1) - u k) * f (u (k + 1))) := by rw [mul_assoc]
+  calc
+    ∑ k in range n, (u (k + 2) - u (k + 1)) * f (u (k + 1)) ≤
+    ∑ k in range n, C * ((u (k + 1) - u k) * f (u (k + 1))) := sum_le_sum this
+    _ = C * ∑ k in range n, (u (k + 1) - u k) * f (u (k + 1)) := mul_sum.symm
+    _ = C * ∑ k in range n, (u (k + 1) - u k) * f (u (k + 1)) := by rw [mul_comm]
+    _ ≤ C * ∑ k in Ico (u 0 + 1) (u n + 1), f k := mul_le_mul_of_nonneg_left
+      (sum_schlomilch_le' hf h_pos hu n) (le_of_lt hCpos)
 end Finset
 
 namespace ENNReal
 
-variable {f : ℕ → ℝ≥0∞}
-
-theorem le_tsum_condensed (hf : ∀ ⦃m n⦄, 0 < m → m ≤ n → f n ≤ f m) :
-    ∑' k, f k ≤ f 0 + ∑' k : ℕ, 2 ^ k * f (2 ^ k) := by
-  rw [ENNReal.tsum_eq_iSup_nat' (Nat.tendsto_pow_atTop_atTop_of_one_lt _root_.one_lt_two)]
-  refine' iSup_le fun n => (Finset.le_sum_condensed hf n).trans (add_le_add_left _ _)
-  simp only [nsmul_eq_mul, Nat.cast_pow, Nat.cast_two]
+variable {u : ℕ → ℕ} {f : ℕ → ℝ≥0∞}
+
+theorem strict_monotone_atTop_atTop (strict_mono : ∀ ⦃m n⦄, 0 ≤ m → m < n → u m < u n) :
+    Tendsto u atTop atTop := by
```

Comment:
I know this file didn't phrase things in terms of [Monotone](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Monotone/Basic.html#Monotone), [StrictMono](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Monotone/Basic.html#StrictMono), [MonotoneOn](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Monotone/Basic.html#MonotoneOn) or [StrictMonoOn](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Monotone/Basic.html#StrictMonoOn), but it probably would be nice if it did. I encourage you to rephrase this file in that language.

If you use those then this lemma is just [StrictMono.tendsto_atTop](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/AtTopBot.html#StrictMono.tendsto_atTop).